plugins/org.jboss.tools.openshift.core/src/org/jboss/tools/openshift/core/server/OpenShiftServerUtils.java

```diff
@@ -404,27 +420,54 @@ private static String getProjectAttribute(String name, String defaultValue, IPro
 		return ServerUtils.getProjectAttribute(name, defaultValue, SERVER_PROJECT_QUALIFIER, project);
 	}
 	
-	public static IDeploymentConfig getDeploymentConfig(IServerAttributes attributes) {
-		IService service = getService(attributes);
-		//TODO use annotations instead of labels
+	/**
```

Comment:
main change is here (changes in other classes are test).
changed the method to throw exceptions so that we can distinguish failure causes properly.